src/App.jsx

```diff
@@ -14,33 +15,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [isSortedAlphabetically, setIsSortedAlphabetically] = useState(false);
+  const [isSortedByLength, setIsSortedByLength] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.localeCompare(b));
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    setGoods(sortedGoods);
+    setIsSortedAlphabetically(true);
+    setIsSortedByLength(false);
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+
+    setGoods(sortedGoods);
+    setIsSortedByLength(true);
+    setIsSortedAlphabetically(false);
+    setIsReversed(false);
+  };
+
```

Comment:
you can simplify this logic just creating one function that will return a sorted array
```
const visibleGoods = getVisibleGoods(goodsFromServer, sortType, isReversed)
```